numpy/random/_pcg64.pyi

```diff
@@ -0,0 +1,27 @@
+from typing import Dict, Literal, Union
+
+from numpy.random.bit_generator import BitGenerator as BitGenerator
+from numpy.random.bit_generator import SeedSequence as SeedSequence
+from numpy.typing import _ArrayLikeInt_co
+
+class PCG64(BitGenerator):
+    def __init__(
+        self, seed: Union[None, int, _ArrayLikeInt_co, SeedSequence] = ...
+    ) -> None: ...
+    def jumped(self, jumps=1) -> PCG64: ...
+    @property
+    def state(
+        self,
+    ) -> Dict[
+        Literal["bit_generator", "state", "has_uint32", "uinteger"],
+        Union[str, int, Dict[str, int]],
```

Comment:
Are you aware of the [`TypedDict`](https://www.python.org/dev/peps/pep-0589/)? It's used for representing a dictionary with a fixed set of (literal) keys, each with their own unique value type (similar to `NamedTuple`):

``` python
import sys
from typing import Dict, Union
if sys.version_info >= (3, 8):
    from typing import TypedDict, Literal
else:
    from typing_extensions import TypedDict, Literal

class _StateDict(TypedDict):
    state: int
    inc: int

class _PCG64Dict(TypedDict):
    bit_generator: Literal["PCG64"]
    state: _StateDict
    has_uint32: int
    uinteger: int

class PCG64:
    @property
    def state(self) -> _PCG64Dict: ...
    @state.setter
    def state(self, value: _PCG64Dict) -> None: ...
```

Reply:
I've moved away from this since the specific keys was problematic since the keys were not constant accrosss the implementations.

Reply:
This shouldn't be an issue with concrete classes though, no?
They'll each just need their own unique `TypedDict`.

For abstract methods this is definitely a bit more challenging as `TypedDict` is, annoyingly enough, not a subtype of `Dict[str, Any]`. As a workaround you can still mark them as `Mapping[str, Any]` though.

Reply:
If you feel creating all the typeddicts is too much work for now, then I'd suggest simply changing the value type to `Any`. `Union`s can get a bit problematic when incorporated in return-types, especially in combination with invariant objects like dictionaries.

Reply:
I've gone with `TypedDicts`. Work well once `Mapping[str, Any]` was used in the base class.